ldap/servers/slapd/back-ldbm/vlv.c

```diff
@@ -377,6 +377,49 @@ vlv_close(ldbm_instance *inst)
     }
 }
 
+/*
+ * List vlv filenames without acceding to the vlv target
+ *  backend (unlike vlv_init).
+ * Note that vlv configuration is not fully checked so it is
+ * possible to get names that are not assiciated to working index.
+ * (Useless empty mdb db files are created in such case)
+ */
+char **
+vlv_list_filenames(ldbm_instance *inst)
+{
+    /* The FE DSE *must* be initialised before we get here */
+    const char *indexfilter = "(objectclass=vlvindex)";
+    char * attrs[] = { (char*)type_vlvName, NULL };
+    Slapi_Entry **entries = NULL;
+    Slapi_PBlock *tmp_pb;
+    char *basedn = NULL;
+    char **names = NULL;
+
+    if (!inst) {
+        return names;
+    }
+
+    basedn = slapi_create_dn_string("cn=%s,cn=%s,cn=plugins,cn=config",
+                                        inst->inst_name, inst->inst_li->li_plugin->plg_name);
+    if (NULL == basedn) {
+        return names;
+    }
+
+    tmp_pb = slapi_search_internal(basedn, LDAP_SCOPE_SUBTREE, indexfilter, NULL, attrs, 0);
+    slapi_pblock_get(tmp_pb, SLAPI_PLUGIN_INTOP_SEARCH_ENTRIES, &entries);
+    for (size_t i = 0; entries && entries[i] != NULL; i++) {
+        const char *name = slapi_entry_attr_get_ref(entries[0], type_vlvName);
```

Comment:
is it entries[0] or entries[i] ?

Reply:
Good catch: It is entry [i] (I am a bit surprised that the freeipa test with multiple vlv index is passing ! )